### PR TITLE
Add support for tar.xz archive file formats

### LINKED ISSUE #708: Use tar.xz archives for Linux if available by default
As the following blog post explains the downloadable builds for Firefox switched from `tar.bz2` to `tar.xz`. We should update mozdownload to support this file extension and at least for newer builds prefer this extension over `tar.bz2`.

https://blog.nightly.mozilla.org/2024/11/28/announcing-faster-lighter-firefox-downloads-for-linux-with-tar-xz-packaging/

### PR DESCRIPTION
**Pull request by whimboo**
Fixes #708.

### CODE REVIEW

File: mozdownload/scraper.py [hunk 1]
```diff
@@ -364,6 +365,14 @@ def _download():
 
         return self.filename
 
+    def get_file_extension(self, binary):
+        suffixes = Path(binary).suffixes
+        if suffixes[-1] in [".bz2", ".xz"]:
```
Comment (whimboo) [inline mozdownload/scraper.py:370]: Actually we also need to support `.tar.gz` which I missed. Also we should change the logic here given that `.tar` isn't required to be always present.

File: mozdownload/scraper.py [hunk 2]
```diff
@@ -68,9 +69,9 @@
                            'android-armeabi-v7a': 'apk',
                            'android-x86': 'apk',
                            'android-x86_64': 'apk',
-                           'linux': 'tar.bz2',
-                           'linux64': 'tar.bz2',
-                           'linux-arm64': 'tar.bz2',
+                           'linux': 'tar.(bz2|xz)',
+                           'linux64': 'tar.(bz2|xz)',
+                           'linux-arm64': 'tar.(bz2|xz)',
```
[the remark this answers is not in the document]
Comment (whimboo) [reply mozdownload/scraper.py:74]: We usually only allow to download the binaries. If you want to download something else then you have to use the `--extension` argument. But I'm as well rewriting this code right now.

File: mozdownload/scraper.py (hunk 1 above)
Comment (whimboo) [reply mozdownload/scraper.py]: This was kinda crappy. With the updated code we do not need this anymore.

**Review by juliandescottes — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**